contrib/aws/internal/eventbridge/eventbridge_test.go

```diff
@@ -0,0 +1,194 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016 Datadog, Inc.
+
+package eventbridge
+
+import (
+	"context"
+	"encoding/json"
+	"strconv"
+	"strings"
+	"testing"
+
+	"github.com/aws/aws-sdk-go-v2/aws"
+	"github.com/aws/aws-sdk-go-v2/service/eventbridge"
+	"github.com/aws/aws-sdk-go-v2/service/eventbridge/types"
+	"github.com/aws/smithy-go/middleware"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"gopkg.in/DataDog/dd-trace-go.v1/ddtrace/mocktracer"
+	"gopkg.in/DataDog/dd-trace-go.v1/ddtrace/tracer"
+)
+
+func TestEnrichOperation(t *testing.T) {
+	mt := mocktracer.Start()
+	defer mt.Stop()
+
+	ctx := context.Background()
+	_, ctx = tracer.StartSpanFromContext(ctx, "test-span")
+
+	input := middleware.InitializeInput{
+		Parameters: &eventbridge.PutEventsInput{
+			Entries: []types.PutEventsRequestEntry{
+				{
+					Detail:       aws.String(`{"key": "value"}`),
+					EventBusName: aws.String("test-bus"),
+				},
+				{
+					Detail:       aws.String(`{"another": "data"}`),
+					EventBusName: aws.String("test-bus-2"),
+				},
+			},
+		},
+	}
+
+	EnrichOperation(ctx, input, "PutEvents")
+
+	params, ok := input.Parameters.(*eventbridge.PutEventsInput)
+	require.True(t, ok)
+	require.Len(t, params.Entries, 2)
+
+	for _, entry := range params.Entries {
+		var detail map[string]interface{}
+		err := json.Unmarshal([]byte(*entry.Detail), &detail)
+		require.NoError(t, err)
+
+		assert.Contains(t, detail, datadogKey)
+		ddData, ok := detail[datadogKey].(map[string]interface{})
+		require.True(t, ok)
+
```

Comment:
We should check in here to make sure that the user defined keys are still present.